warehouse/query-core/src/main/java/datawave/next/ListDocIdIterator.java

```diff
@@ -0,0 +1,177 @@
+package datawave.next;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.commons.jexl3.parser.JexlNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+
+import datawave.core.iterators.DatawaveFieldIndexListIteratorJexl;
+import datawave.query.Constants;
+import datawave.query.jexl.nodes.ExceededOr;
+import datawave.query.jexl.nodes.QueryPropertyMarker;
+import datawave.query.jexl.nodes.QueryPropertyMarker.MarkerType;
+
+/**
+ * An alternate implementation of the {@link DatawaveFieldIndexListIteratorJexl}
+ * <p>
+ * Sorted order is not guaranteed.
+ */
+public class ListDocIdIterator extends DocIdIterator {
+
+    private static final Logger log = LoggerFactory.getLogger(ListDocIdIterator.class);
+
+    private List<String> values;
+    private List<Range> ranges;
+    private Range currentRange = null;
+
+    /**
+     * Minimal constructor requires the source, row, and JexlNode
+     *
+     * @param source
+     *            the source iterator
+     * @param row
+     *            the row
+     * @param node
+     *            the JexlNode
+     */
+    public ListDocIdIterator(SortedKeyValueIterator<Key,Value> source, String row, JexlNode node) {
+        super(source, row, node);
+    }
+
+    @Override
+    protected void parseNode(JexlNode node) {
+        QueryPropertyMarker.Instance instance = QueryPropertyMarker.findInstance(node);
+        Preconditions.checkArgument(instance.getType().equals(MarkerType.EXCEEDED_OR));
+
+        ExceededOr exceededOr = new ExceededOr(instance.getSource());
+        this.field = exceededOr.getField();
+        this.values = new ArrayList<>();
+        this.values.addAll(exceededOr.getParams().getValues());
+    }
+
+    private void buildRanges() {
+        this.ranges = new ArrayList<>();
+        Collections.sort(values);
+        for (String value : values) {
+            if (datatypeFilter != null && !datatypeFilter.isEmpty()) {
+                Key start = new Key(this.row, this.prefix, value + "\u0000" + datatypeFilter.first() + "\u0000");
+                Key end = new Key(this.row, this.prefix, value + "\u0000" + datatypeFilter.last() + "\u0000" + Constants.MAX_UNICODE_STRING);
+                ranges.add(new Range(start, true, end, false));
+            } else {
+                Key start = new Key(this.row, this.prefix, value + "\u0000");
+                Key end = new Key(this.row, this.prefix, value + "\u0000" + Constants.MAX_UNICODE_STRING);
+                ranges.add(new Range(start, true, end, false));
+            }
```

Comment:
seen these ranges generated in a few different classes. Move to common code

Reply:
Yup: https://github.com/NationalSecurityAgency/datawave/issues/2916